### PR TITLE
Add an emacs testdrive mode

### PR DESCRIPTION
**Pull request by quodlibetor**
This uses `polymode` to actually enter the various meaningful submodes for different
parts of testdrive files, which is (so far) a pretty nice experience.

This adds enough syntax highlighting to make testdrive _much_ more pleasant in emacs:

image

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4588)

### CODE REVIEW

File: misc/editor/emacs/mz-testdrive.el [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+;; Copyright Materialize, Inc. All rights reserved.
+;;
+;; Use of this software is governed by the Business Source License
+;; included in the LICENSE file at the root of this repository.
+;;
+;; As of the Change Date specified in that file, in accordance with
+;; the Business Source License, use of this software will be governed
+;; by the Apache License, Version 2.0.
+
+;; Author: Brandon W Maister <bwm@materialize.io
```
Comment (benesch) [inline misc/editor/emacs/mz-testdrive.el:12]: Missing `>`.

**Review by benesch — APPROVED** [review]
I know nothing about emacs, but seems great!

File: misc/editor/emacs/mz-testdrive.el (hunk 1 above)
Comment (quodlibetor) [reply misc/editor/emacs/mz-testdrive.el]: nice catch

**Review by cirego — COMMENTED** [review]
This change *almost* makes me want to use Emacs 😉

**Pull request merged**